CONTRIBUTING.md

```diff
@@ -30,6 +32,52 @@ Once you have chromedriver installed and available in `PATH` you can re-generate
 
 To add a new test case add another HTML file to `/test_fixtures` following the current tests as a template for new tests.
 
+### Writing tests
+
+1. All tests should be wrapped in a module called `tests`, to ensure they are not compiled unless testing
+
+```rs
+#[cfg(test)]
+mod tests {
+    // Place tests here
+}
+```
+
+2. For unit-testing this should be placed within the file of the definition they are testing
+    1. If file becomes significantly long, tests should be split out into its own file under the same module
+
+```rs
+// file: ./src/my_struct.rs
+struct MyStruct;
+
+impl MyStruct {
+    fn some_method() { .. }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_of_some_method() { .. }
+}
+```
+
+4. For integration tests this should be placed within the `./tests` folder
+
+```rs
+// file: ./tests/my_integration_test.rs
+#[cfg(test)]
+mod tests {

```

Comment:
Is it necessary to wrap this in a module again, when it's in a separate file already?
It might also work to use `#![cfg(test)]` at the top of the file instead, but I haven't tested this out.

Reply:
IIRC rust treats a file as a module, in which case wrapping it another module is completely redundant. Good spot :)

Reply:
According to the [rust docs](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory) everything in the `tests` directory will only be compiled on tests, which means we don't need to specify `#![cfg(tests)]`